### ISSUE TITLE
Phiprof workunit support

### ISSUE THREAD

**Issue by lkotipal** [UO-OACISS/tau2#9]
Phiprof has a feature for using workunits (for example cells propagated) along with timers to output statistics such as cells propagated per second. Right now Tau discards workunits on calls to `phiprof::stop()`. Is it possible to collect workunits as a metric for analysis in ParaProf and PerfExplorer?

**Comment by khuck** [comment]
Yes, I think that should be doable. I'll look into it and see how we can support that...
